.github/workflows/build.yml

```diff
@@ -0,0 +1,121 @@
+name: Build
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+jobs:
+  prep:
+    name: Prepare the build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Update the 'latest' tag
+        if: github.ref == 'refs/heads/main'
+        run: |
+           git config user.name github-actions
+           git config user.email github-actions@github.com
+           git push origin :refs/tags/latest || true
+           git tag -m latest -fa latest || true
+           git push origin main --tags
+  build:
+    name: Build on ${{ matrix.os }} with Java ${{ matrix.java }}
+    needs: prep
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        java: [16, 11, 8, 17-ea]
+        exclude:
+          - os: macos-latest
+            java: 8
+          - os: macos-latest
+            java: 11
+          - os: macos-latest
+            java: 17-ea
+          - os: windows-latest
+            java: 8
+          - os: windows-latest
+            java: 11
+          - os: windows-latest
+            java: 17-ea
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Java JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: adopt
+          java-version: ${{ matrix.java }}
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Update submodules
+        run: python checker.py update
+      - name: Install dependencies
+        run: python checker.py dldeps
+      - name: Build
+        run: |
+          python checker.py build
+          python checker.py jar
+          python checker.py image
+          python checker.py war
+      - name: Test
+        shell: bash
```

Comment:
I think this wasn't running for Windows before, but I guess it's OK with the explicit `bash` environment?

Reply:
> I think this wasn't running for Windows before, but I guess it's OK with the explicit `bash` environment?

Right, yes. And I guess running it in bash on Windows is kind of not actually modeling/testing what a normal user in their normal environment on Windows would be doing — but anyway that’s only the case for this one test step, which itself is anyway testing the runtime behavior and not just the build success. So it seems OK to use bash on Windows for this one part.